File: HealthEase/HealthEase.Services/Database/Appointment.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HealthEase.Services.Database
{
    public partial class Appointment : BaseEntity
    {
        [Key]
        public int AppointmentId { get; set; }

        [Required]
        public DateTime AppointmentDate { get; set; }

        [Required]
        public TimeSpan AppointmentTime { get; set; }

        public string Status { get; set; } = null!;

        public string? StatusMessage { get; set; }

        public string? Note { get; set; }

        public bool IsPaid { get; set; }

        public DateTime? PaymentDate { get; set; }

        public int DoctorId { get; set; }
        [ForeignKey("DoctorId")]
        public virtual Doctor Doctor { get; set; } = null!;

        public int PatientId { get; set; }
        [ForeignKey("PatientId")]
        public virtual Patient Patient { get; set; } = null!;
        public int AppointmentTypeId { get; set; }
        [ForeignKey("AppointmentTypeId")]
        public virtual AppointmentType AppointmentType { get; set; } = null!;
    }
}

File: HealthEase/UI/healthease_mobile/lib/models/appointment.dart
import 'package:healthease_mobile/models/appointment_type.dart';
import 'package:healthease_mobile/models/doctor.dart';
import 'package:healthease_mobile/models/patient.dart';
import 'package:json_annotation/json_annotation.dart';
part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  int? appointmentId;
  String? appointmentDate;
  String? appointmentTime;
  String? status;
  String? statusMessage;
  String? note;
  bool? isPaid;
  String? paymentDate;
  int? doctorId;
  int? patientId;
  int? appointmentTypeId;
  Doctor? doctor;
  Patient? patient;
  AppointmentType? appointmentType;

  Appointment();

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
